src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -306,6 +310,56 @@ private void validateSsl() {
     }
   }
 
+  private void validateVersion(RestHighLevelClient client) {
+    if (!config.isDataStream()) {
+      return;
+    }
+    MainResponse response;
+    try {
+      response = client.info(RequestOptions.DEFAULT);
+    } catch (IOException e) {
+      // Same error messages as from validating the connection.
+      return;
+    }
+    String versionNumber = response.getVersion().getNumber();
+    if (compareVersions(versionNumber, DATA_STREAM_COMPATIBLE_ES_VERSION) < 0) {
+      String errorMessage = String.format(
+          "Elasticsearch version %s is not compatible with data streams. Elasticsearch"
+              + "version must be at least %s.",
+          versionNumber,
+          DATA_STREAM_COMPATIBLE_ES_VERSION
+      );
+      addErrorMessage(CONNECTION_URL_CONFIG, errorMessage);
+      addErrorMessage(DATA_STREAM_TYPE_CONFIG, errorMessage);
+      addErrorMessage(DATA_STREAM_DATASET_CONFIG, errorMessage);
+    }
+  }
+
+  /**
+   * Compares <code>versionNumber</code> to <code>compatibleVersion</code>. Ignores
+   * the patch version number.
+   *
+   * @return  a negative integer, zero, or a positive integer if
+   *          <code>versionNumber</code> is less than, equal to, or greater
+   *          than <code>compatibleVersion</code>.
+   */
+  public int compareVersions(String versionNumber, String compatibleVersion) {
```

Comment:
```suggestion
  private int compareVersions(String versionNumber, String compatibleVersion) {
```

Reply:
Oops, forgot to change it back after I played around with it. Thanks!